### DISCUSSION TITLE
Change UI style of the nowcasting chart

### DISCUSSION THREAD

**Discussion by humpyalpha** [breezy-weather/breezy-weather#1309] [Ideas]
### Describe your idea

The current precipitation histogram's legend font overlaps, and the chart style clearly differs from the overall page style. I hope it can be improved.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
If you use line charts once, why would you need line charts everywhere? Most of the time, we have a mix. And we already have bar charts with air quality and UV.

Also, bar charts are common for nowcasting, because line charts are not fitted when you have a lot of data to put. Feel free to try to replace the bar chart with a line chart, you will see how ugly and unreadable it is.

**Comment by humpyalpha** [comment]
Thank you for your hard work! I don't have a strong preference between using a bar chart or a line chart. My main concern is ensuring a consistent UI style across the software. Even with bar charts, the style of the precipitation chart seems somewhat different from the others. Would it be possible for you to consider aligning the UI style of the precipitation chart with the rest? I really appreciate your understanding and help!



获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by papjul** [reply]
Can you clarify what is different/non consistent exactly? We use the same colors for thresholds line, same dynamic color for the bar colors, etc.

Do you have a pull request in mind?

**Reply by humpyalpha** [reply]
Oh, my apologies! I originally attached two images to the email, but I didn’t realize that attachments aren’t visible on GitHub. I'm sending them to you now.

Hearing what you said, I’m starting to wonder if I might have encountered a bug. If it is indeed a bug, my device is a Xiaomi 13 running Xiaomi HyperOS 1.0.8.0.UMCCNXM.
![微信图片_20240916163329](https://github.com/user-attachments/assets/f353ad90-17f2-4f94-a4ed-bbbd5961b8a2)
![微信图片_20240916163333](https://github.com/user-attachments/assets/853abce5-4b9c-4add-b077-b53bc0f3b4d2)
As you can see, the precipitation bar chart on my end has issues with overlapping text and inconsistent styling. Additionally, the option I set in the global settings to disable the threshold line hasn’t taken effect in the precipitation chart.

**Reply by papjul** [reply]
Yes, they are indeed bugs. Can you report:
- One bug for the overlapping texts
- One bug for the threshold lines appearing even if it was disabled

As for the style, it's the style from the current condition (blue background). If it was cloudy at that time, the bar colors would have been grey like the grey background.

**Reply by papjul** [reply]
Converted into #1310 and #1311; closing
